### ISSUE TITLE
New rule: no-bind

### ISSUE THREAD

**Issue by huashiyiqike** [palantir/tslint-react#44]
To circumvent re-rendering by creating a new  function each time when binding this to functions

[https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md)

```
"jsx-no-bind": [<enabled>, {
  "ignoreRefs": <boolean> || false,
  "allowArrowFunctions": <boolean> || false,
  "allowBind": <boolean> || false
}]
```

**Comment by tp** [comment]
I found that the performance impact may not even be the real issue this would be helpful with, but rather TypeScript's lenient typing of bind (`bind(this: Function, thisArg: any, ...argArray: any[]): any;`), which results in parameter-mismatches not being caught.

**Comment by huashiyiqike** [comment]
@tp Great point! I didn't notice that, here is the link [https://basarat.gitbooks.io/typescript/content/docs/tips/bind.html](https://basarat.gitbooks.io/typescript/content/docs/tips/bind.html)

**Comment by adidahiya** [comment]
will be available in v2.6.0
